src/main/java/ch/njol/skript/expressions/ExprTool.java

```diff
@@ -102,35 +107,38 @@ public void setItem(final @Nullable ItemStack item) {
 						};
 					}
 				}
-				final EntityEquipment e = p.getEquipment();
-				if (e == null)
+				final EntityEquipment eq = ent.getEquipment();
+				if (eq == null)
 					return null;
-				return new EquipmentSlot(e, EquipmentSlot.EquipSlot.TOOL) {
+				return new EquipmentSlot(eq, offHand ? EquipmentSlot.EquipSlot.OFF_HAND : EquipmentSlot.EquipSlot.TOOL) {
 					@Override
 					public String toString(@Nullable Event event, boolean debug) {
-						return (getTime() == 1 ? "future " : getTime() == -1 ? "former " : "") + Classes.toString(getItem());
+						String time = getTime() == 1 ? "future " : getTime() == -1 ? "former " : "";
```

Comment:
I don't think you need to take into account the time as this is part of the ExprTime

Reply:
Good to know, will remove this.